RevenueCatUI/CustomerCenter/ViewModels/BaseManageSubscriptionViewModel/BaseManageSubscriptionViewModel.swift

```diff
@@ -0,0 +1,337 @@
+    private(set) var purchasesProvider: CustomerCenterPurchasesType
+
+    init(
+        screen: CustomerCenterConfigData.Screen,
+        actionWrapper: CustomerCenterActionWrapper,
+        purchaseInformation: PurchaseInformation? = nil,
+        refundRequestStatus: RefundRequestStatus? = nil,
+        purchasesProvider: CustomerCenterPurchasesType,
+        loadPromotionalOfferUseCase: LoadPromotionalOfferUseCaseType? = nil) {
+            self.screen = screen
+            self.paths = screen.filteredPaths
+            self.purchaseInformation = purchaseInformation
+            self.purchasesProvider = purchasesProvider
+            self.refundRequestStatus = refundRequestStatus
+            self.actionWrapper = actionWrapper
+            self.loadPromotionalOfferUseCase = loadPromotionalOfferUseCase
+            ?? LoadPromotionalOfferUseCase(purchasesProvider: purchasesProvider)
+            self.restoreAlertType = .loading
+        }
+
+#if os(iOS) || targetEnvironment(macCatalyst)
+    func determineFlow(for path: CustomerCenterConfigData.HelpPath, activeProductId: String? = nil) async {
+        // Convert the path to an appropriate action using the extension
+        if let action = path.asAction() {
+            // Send the action through the action wrapper
+            self.actionWrapper.handleAction(.buttonTapped(action: action))
+        }
+
+        switch path.detail {
+        case let .feedbackSurvey(feedbackSurvey):
+            self.feedbackSurveyData = FeedbackSurveyData(
+                configuration: feedbackSurvey,
+                path: path) { [weak self] in
+                    Task {
+                        await self?.onPathSelected(path: path)
+                    }
+                }
+
+        case let .promotionalOffer(promotionalOffer) where purchaseInformation?.store == .appStore:
+            if promotionalOffer.eligible {
+                self.loadingPath = path
+                let result = await loadPromotionalOfferUseCase.execute(promoOfferDetails: promotionalOffer)
+                switch result {
+                case .success(let promotionalOfferData):
+                    self.promotionalOfferData = promotionalOfferData
+                case .failure:
+                    await self.onPathSelected(path: path)
+                    self.loadingPath = nil
+                }
+            } else {
+                Logger.debug(Strings.promo_offer_not_eligible_for_product(
+                    promotionalOffer.iosOfferId, activeProductId ?? ""
+                ))
+                await self.onPathSelected(path: path)
+            }
+
+        default:
+            await self.onPathSelected(path: path)
+        }
+    }
+
+    func onDismissInAppBrowser() {
+        self.inAppBrowserURL = nil
+    }
+
+    func openManagementURL(for productId: String) {
+        guard let purchaseInformation,
+              purchaseInformation.productIdentifier == productId,
```

Comment:
Is there an edge case here if there are two products with the same `productId` but different stores? Meaning if we should also check for a matching store here

Reply:
Oh good catch. This is actually not used anymore. I'll delete it.
I am using:

```swift
        case .cancel where purchaseInformation?.store != .appStore:
            if let url = purchaseInformation?.managePurchaseURL {
                self.inAppBrowserURL = IdentifiableURL(url: url)
            } else {
                // WIP: shor error alert?
            }
```

Reply:
Just added a test for this, we're good